### ISSUE TITLE
Package request: piped

### ISSUE THREAD

**Issue by Atemu** [NixOS/nixpkgs#270008]
**Project description**

A self-hosted alternative youtube back-end.

It uses pnpm for building which is not supported in Nixpkgs properly yet: https://github.com/NixOS/nixpkgs/issues/231513

**Metadata**

* homepage URL: https://github.com/TeamPiped/Piped/
* source URL: https://github.com/TeamPiped/Piped/
* license: agpl3
* platforms: all

**Priorities**

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by squalus** [comment]
I have a flake here you might find useful: https://github.com/squalus/piped-flake

**Comment by Atemu** [comment]
Thanks. However you have licensed your project under the AGPL which means that it can't be incorporated into Nixpkgs (MIT).

You're of course free to do that but I think it's better to license Nix expressions under MIT because of this issue.

**Comment by squalus** [comment]
> Thanks. However you have licensed your project under the AGPL which means that it can't be incorporated into Nixpkgs (MIT).
> 
> You're of course free to do that but I think it's better to license Nix expressions under MIT because of this issue.

Changed it to 0bsd.

**Comment by Atemu** [comment]
Oh. It appears we duplicated effort @Defelo. Do you mind if I cherry-pick some of your stuff into my branch?

**Comment by Defelo** [comment]
No, sure, go ahead. I didn't know you were also working on this, when I created my PR a month ago.

**Comment by Atemu** [comment]
It's always good to ping the packaging request when something relevant happens such as a PR. I was watching this thread and would have noticed.

I have refactored it a bit in https://github.com/NixOS/nixpkgs/pull/326238, could you take a look? (I will squash the commits later.)

**Comment by Atemu** [comment]
Please always mention why you close issues. I assume it's due to https://github.com/NixOS/nixpkgs/pull/354259 which was never linked here.

That PR however does not do anything useful. It's just the front-end which ...doesn't do anything. It's just website assets.

I wonder why it was merged at all? cc @thiagokokada 

We need the whole thing for it to be of any use.

**Comment by thiagokokada** [comment]
> Please always mention why you close issues. I assume it's due to #354259 which was never linked here.
> 
> That PR however does not do anything useful. It's just the front-end which ...doesn't do anything. It's just website assets.
> 
> I wonder why it was merged at all? cc @thiagokokada
> 
> We need the whole thing for it to be of any use.

Without entering in another discussion of policies inside nixpkgs, I generally assume if someone is packaging something in nixpkgs, they found it useful in their current state.

@lucasew considering that #354259 is just the frontend, why it is useful?

Anyway, just saw your PR @Atemu, I think you can just rebase it accordingly to the #354259 changes (like rename `piped-frontend` to `piped`, but keep your changes).
